manticore/core/manticore.py

```diff
@@ -365,6 +351,8 @@ def __init__(self, initial_state, workspace_url=None, outputspace_url=None, **kw
         # Note that each callback will run in a worker process and that some
         # careful use of the shared context is needed.
         self.plugins = set()
+        self._introspection_plugin: type = IntrospectionAPIPlugin
```

Comment:
> Maybe .plugins should be a dict[name -> plugin]

This is fine with me. It would simplify the introspection API _and_ the TUI infrastructure. ManticoreEVM access the plugins list directly in a few places, but we can refactor that.

> What IntrospectionAPIPlugin need to do that can not be done via an event via a plain Plugin

It's a normal plugin. It doesn't have any privileged APIs or use the daemon thread interface.

So we're clear though, `_introspection_plugin` is the _type_, not the plugin itself. The plugin instance is constructed later, and lives under `._introspector`

Reply:
~Oh right this it is a type. So why do we need to save this type here?~
So we construct it in a later time.

So why can not a normal plugin be created like any other plugin from the main script then added somehow?

Reply:
From a technical standpoint, it's a normal plugin, so there would be nothing wrong with creating it outside of Manticore and passing it to `register_plugin`. But since it's necessary to drive the introspection API, I've privileged it so that it's always created as part of `__init__`

Reply:
Nice! I think ALL plugins should be registered at init! And that `register_plugin` should be deprecated.

So the only question I have is why manticore needs an special internal reference to a plugin....